test/runtests.jl

```diff
@@ -128,8 +128,17 @@ end
         drfl_min = DatetimeRotatingFileLogger(dir, raw"\m\i\n-YYYY-mm-dd-HH-MM.\l\o\g")
         func = (io, args) -> println(io, reverse(args.message))
         drfl_fmt = DatetimeRotatingFileLogger(func, dir, raw"\f\m\t-YYYY-mm-dd-HH-MM-SS.\l\o\g")
+        function compressor(file)
+            open(file) do src; open(file * ".compressed", "w") do sink
+                for line in eachline(src)
+                    write(sink, replace(line, r"Info" => "I"))
+                end
+            end end
```

Comment:
I think for testing purposes we should just set a variable in the parent scope for what the filename was,
and then check after that it was called on all the right filenames.

Reply:
I don't follow.

Reply:
as in we have code that is

```julia

@testset "..." begin
    ...
    callback_record = []
    drfl_cb = DatetimeRotatingFileLogger(
        dir, raw"\c\o\m\p-YYYY-mm-dd-HH-MM-SS.\l\o\g";
        rotation_callback=x->push!(callback_record, x)
    )
    ...
    ...
    # Make sure the rotation callback was triggered
    @test length(callback_record) == 2
    # + what ever other test wanted
end
```

This is easier to test than a more real-world use-case.
But still proves the components are interacting as we expect.